packages/graphql-yoga/__tests__/request-cancellation.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { createSchema, createYoga } from '../src/index';
+
+describe('request cancellation', () => {
+  it('request cancellation stops invocation of subsequent resolvers', async () => {
+    const rootResolverGotInvokedD = createDeferred();
+    const requestGotCancelledD = createDeferred();
+
+    let aResolverGotInvoked = false;
+    let rootResolverGotInvoked = false;
+
+    const schema = createSchema({
+      typeDefs: /* GraphQL */ `
+        type Query {
+          root: A!
+        }
+
+        type A {
+          a: String!
+        }
+      `,
+      resolvers: {
+        Query: {
+          async root() {
+            rootResolverGotInvoked = true;
+            rootResolverGotInvokedD.resolve();
+            await requestGotCancelledD.promise;
+
+            return { a: 'a' };
+          },
+        },
+        A: {
+          a() {
+            aResolverGotInvoked = true;
+            return 'a';
+          },
+        },
+      },
+    });
+
+    const yoga = createYoga({ schema });
+
+    const abortController = new AbortController();
+    const promise = Promise.resolve(
+      yoga.fetch('http://yoga/graphql', {
+        method: 'POST',
+        body: JSON.stringify({ query: '{ root { a } }' }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        signal: abortController.signal,
+      }),
+    );
+
+    await rootResolverGotInvokedD.promise;
+    abortController.abort();
+
+    requestGotCancelledD.resolve();
+
+    // The request was aborted, so the promise should raise instead
+    const result = await promise.then(res => res.text());
+    // why does this return a (invalid) GraphQL response?
+    expect(result).toMatchInlineSnapshot(`"{"data":{"root":{"a":null}}}"`);
```

Comment:
@ardatan I think if the execution is aborted the Promise returned from `yoga.fetch` should reject with an "Request got aborted" error.

Also, the executor should raise an error instead of returning null for the field processed right now.

I added these snapshots to showcase the invalid behavior.